internal/catchup_push_handler.go

```diff
@@ -0,0 +1,43 @@
+package internal
+
+import (
+	"wal-g/utility"
+
+	"github.com/wal-g/tracelog"
+)
+
+// HandleCatchupPush is invoked to perform wal-g catchup-push
+func HandleCatchupPush(uploader *Uploader, archiveDirectory string) {
```

Comment:
Actually it looks like backup-push. I guess you didn't catch the idea here. This command should be --from-lsn and make an incremental backup from pages which LSN greater than "--from-lsn" and probably put it in a different path at storage for a consistent sequence of backups in general path of the storage